python/ray/experimental/packaging/load_package.py

```diff
@@ -196,6 +195,21 @@ def __repr__(self):
             self._module, self._runtime_env)
 
 
+def _validate_stub_file(stub_file: str):
+    if not os.path.exists(stub_file):
+        raise ValueError("Stub file does not exist: {}".format(stub_file))
+    for line in open(stub_file):
+        line = line.replace("\n", "")
+        if line.startswith("import ") or line.startswith("from "):
+            if line != "import ray" and "noqa" not in line:
+                raise ValueError(
```

Comment:
It would be more robust to use Python's parser for this:
https://docs.python.org/3.7/library/ast.html#ast.parse

Reply:
I tried this out, but it's complicated that I'd have more faith in this existing code to work correctly.